src/main/java/uk/gov/pay/api/model/search/card/PaymentForSearchResult.java

```diff
@@ -85,7 +85,10 @@ public static PaymentForSearchResult valueOf(
                 paymentResult.getMetadata().orElse(null),
                 paymentResult.getFee(),
                 paymentResult.getNetAmount(),
-                paymentResult.getAuthorisationSummary(),
+                paymentResult.getAuthorisationSummary() == null ?
+                        null : paymentResult.getAuthorisationSummary().getThreeDSecure() == null ?
+                        null : paymentResult.getAuthorisationSummary().getThreeDSecure().isRequired() ?
+                        paymentResult.getAuthorisationSummary() : null,
```

Comment:
This is the third time this logic is repeated — could it be encapsulated in a helper method? I think even a static method somewhere would be preferable to this repetition.